stable_baselines3/common/envs/multi_input_envs.py

```diff
@@ -0,0 +1,266 @@
+from typing import Dict, Tuple, Union
+
+import gym
+import numpy as np
+
+from stable_baselines3.common.type_aliases import GymStepReturn
+
+
+class SimpleMultiObsEnv(gym.Env):
+    """
+    Base class for GridWorld-based MultiObs Environments 4x4  grid world.
+
+    .. code-block:: text
+
+        ____________
+       | 0  1  2   3|
+       | 4|¯5¯¯6¯| 7|
+       | 8|_9_10_|11|
+       |12 13  14 15|
+       ¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+
+    start is 0
+    states 5, 6, 9, and 10 are blocked
+    goal is 15
+    actions are = [left, down, right, up]
+
+    simple linear state env of 15 states but encoded with a vector and an image observation
+
+    :param num_col: Number of columns in the grid
+    :param num_row: Number of rows in the grid
+    :param random_start: If true, agent starts in random position
+    :param noise: Noise added to the observations
+    """
+
+    def __init__(
+        self,
+        num_col: int = 4,
+        num_row: int = 4,
+        random_start: bool = True,
+        noise: float = 0.0,
+        discrete_actions: bool = True,
+    ):
+        super(SimpleMultiObsEnv, self).__init__()
+
+        self.vector_size = 5
+        self.img_size = [1, 64, 64]
+
+        self.random_start = random_start
+        self.discrete_actions = discrete_actions
+        if discrete_actions:
+            self.action_space = gym.spaces.Discrete(4)
+        else:
+            self.action_space = gym.spaces.Box(0, 1, (4,))
+
+        self.observation_space = gym.spaces.Dict(
+            spaces={
+                "vec": gym.spaces.Box(0, 1, (self.vector_size,)),
+                "img": gym.spaces.Box(0, 255, self.img_size, dtype=np.uint8),
+            }
+        )
+        self.count = 0
+        self.max_count = 100
+        self.log = ""
+        self.state = 0
+        self.action2str = ["left", "down", "right", "up"]
+        self.noise = noise
+        self.init_possible_transitions()
+
+        self.init_state_mapping(num_col, num_row)
+
+        self.max_state = len(self.state_mapping) - 1
+
+    def random_upsample_img(
+        self,
+        value_range: Tuple[int, int] = (0, 255),
+        initial_size: Tuple[int, int] = (4, 4),
+        up_size: Tuple[int, int] = (64, 64),
+    ) -> np.ndarray:
+        """
+        Generated a random image and upsample it.
+
+        :param value_range: The range of values for the img
+        :param initial_size: The initial size of the image to generate
+        :param up_size: The size of the upsample
+        :return: upsampled img
+        """
+        im = np.random.randint(value_range[0], value_range[1], initial_size, dtype=np.int32)
+        return np.array(
```

Comment:
is that double for loop really needed?

Reply:
No, I was originally going to add something that upsamples a simple grid representation of the grid world but I think there are better options for testing the dict observations.

Reply:
Removed in latest commit.